ext/webextension/src/browser_action/main_popup.js

```diff
@@ -241,21 +241,54 @@ function updateUIForDomainSettings(combined)
 }
 
 function extractDomainFromUrl(url) {
-    if (!url || url.startsWith('about:')
-        || url.startsWith('resource:')
-        || url.startsWith('moz-extension:')
-        || url.startsWith('chrome-extension:')
-        || url.startsWith('chrome:'))
-        url = '';
-    let domain_parts = parseUri(url).domain.split(".");
-    let significant_parts = 2;
-    const common_slds = ['co','com','gov','govt','net','org','edu','priv','ac'];
-    let second_level_domain = domain_parts[domain_parts.length-2].toLowerCase();
-    if (domain_parts.length > 2 && common_slds.includes(second_level_domain))
-        significant_parts = 3;
-
-    let significant_domain = domain_parts.slice(-significant_parts)
-    return [domain_parts, significant_domain];
+    if (!url)
+        throw "extractDomainFromUrl: No url provided!";
```

Comment:
Throwing here is not such a great Idea.. everything that follows `extractDomainFromUrl` will be skipped, in particular loadSites and updateUIForDomainSettings. 
So instead of having access to all your sites, it appears to have lost all data.

probably better to return some sane default